### DISCUSSION TITLE
Chained pages

### DISCUSSION THREAD

**Discussion by Limezy** [athensresearch/athens#1029] [Feature Requests]
Even if I don't use the daily pages, I like the look and feel of them.
They look like "chained", because if you scroll up, the next one (that is, the day before) will show up. 
That's great, but why should it be only for daily pages ?

### Basic concept

Some folks out there are using a lot the so called "Zettelkasten" method, which is basically like a chain of notes, but made in a way that you can always add a new note where you want inside the chain, using a kind of tree nomenclature, as explained quite well in that image : 

![image](https://user-images.githubusercontent.com/24638389/115906518-31a8d880-a49a-11eb-99a1-41ce16c6c879.png)

(**credits to :** https://www.lesswrong.com/posts/NfdHG6oHBJ8Qxc26s/the-zettelkasten-method-1)

I think we could support and attract a part of the Zettelkasten community with a support for such "chained pages", or "chained cards", whatever we want to call them.

The idea would be for example to introduce a way to add a chain name and number to a page. 
(The syntax below is just a quick proposal and may be optimized of course.)

Let's say, for the sake of that discussion, that I type `[[Page Title]]~chain`
I've just created a page `[[Page Title]]` and placed it at the top of a chain `chain`.

If I have another page that I want to link with that first one, I just have to add `~chain~1` : `[[Next idea]]~chain~1`
After having written my `[[Third idea]]~chain~2`, I realize I want to add a new though between my pages 1 and 2 from the chain, so I just add a new page `[[Inserted idea]]~chain~1A`

Then I want to start a new line of though, `[[New start]]~otherchain` and so on... 

### How the UX would work : 

The idea is that you can then scroll through a chain just like scrolling through your daily notes. 
Just open one of the pages from the chain (let's say the `[[Inserted idea]]~chain~1A`). It will look like a card, just like for the daily notes. I can scroll up and will find the previous one in the tree, that is `[[Next idea]]~chain~1`. I can scroll down and I will find the next one in the tree, that is `[[Third idea]]~chain~2` etc... 

The UX could also allow me to "insert" a new page in my chain. Let's say I'm looking at page `[[Inserted idea]]~chain~1A`. I click "insert chained page after", and that creates me automatically a `[[  ]]~chain~1B` page on which I can fill a title and start writing !

Obviously, the chains would support autocompletion, so let's say I'm writing on a `[[My idea]]` page, if I type `~` after the title it will provide me with a choice of all existing chains. Once I click on "chain" to insert the `[[My idea]]` page into the chain `chain`, I can chose where to put it in the chain

Backlink filters should remain while scrolling through a page (see below an example why)

### Back to daily pages

In the end, daily pages would be nothing else than a special chain.
But we could imagine (or create manually) monthly pages, annually pages.
For example, I use a lot the dates with only the year number in my Athens. I would love to have an automatic chain, which would make my years scrollable. If I go to [[2020]], then suddenly I want to see what was published by the same author the previous or next years, I just filter the backlinks, then scroll up or down through the years and done !

**Comment by jefftangx** [comment]
In addition to vertical chains, there could also be horizontal chains ~> https://github.com/azlen/roam-themes
Perhaps both, supported on a sort of Infinite Canvas #968 ~> https://twitter.com/azlenelza/status/1272600877493137408
